soc/arm/nordic_nrf/nrf91/Kconfig.series

```diff
@@ -0,0 +1,26 @@
+# Kconfig.series - Nordic Semiconductor nRF91 MCU line
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config  SOC_SERIES_NRF91X
+	bool "Nordic Semiconductor nRF91 series MCU"
+	select CPU_CORTEX_M33
+	select CPU_HAS_ARM_MPU
+	select CPU_HAS_NRF_IDAU
+	select CPU_HAS_FPU
+	select ARMV8_M_DSP
+	select SOC_FAMILY_NRF
+	select NRF_RTC_TIMER
+	select CLOCK_CONTROL
+	select CLOCK_CONTROL_NRF5
```

Comment:
Why you need CLOCK_CONTROL when you are requesting CLOCK_CONTROL_NRF5?
Is there a missing requirement in CLOCK_CONTROL_NRF5?

Reply:
nRF51 and nRF52 SOCs (and simulated nRF boards) select both Kconfig options - this is perhaps a bit problematic, but cannot be solved under this PR. We should need a separate pull request for that. Would you be OK with this? I can track it under #12199

Reply:
@pizi-nordic @agross-linaro I did some investigation here:
- In some SOC/SOC_SERIES (e.g. NXP),the vendor-specific option is defined in the "defconfig" file, and set to "default=y" if CLOCK_CONTROL is selected already (at SOC level)
- In several SOC/SOC_SERIES (e.g. STM32) CLOCK_CONTROL_<vendor-specific> is _selected_ conditionally,if CLOCK_CONTROL is selected (normally) at the BOARD level.
For example `select CLOCK_CONTROL_STM32_CUBE if CLOCK_CONTROL`
- For Nordic nRFx, CLOCK_CONTROL is directly selected at the SOC_SERIES_NRFxx. That is why we only need:  `select CLOCK_CONTROL_NRF5`, i.e. we can skip the conditional (if CLOCK_CONTROL)

Conclusions:
1. It is the other way @agross-linaro , i.e. CLOCK_CONTROL selects the vendor-specific option
2. I can make this look like the STM32 case, i.e. `select CLOCK_CONTROL_NRF5 if CLOCK_CONTROL` but it is not really needed since CLOCK_CONTROL is selected under the same symbol. So I would rather leave it as is.


---

We _do_ have an issue with naming, here, since CLOCK_CONTROL_NRF5 is "used" in nRF91 series. I think we can  simply rename CLOCK_CONTRO_NRF5 --> CLOCK_CONTROL_NRF. I think I will open a separate PR for this.

Reply:
>We do have an issue with naming, here, since CLOCK_CONTROL_NRF5 is "used" in nRF91 series. I think we can simply rename CLOCK_CONTRO_NRF5 --> CLOCK_CONTROL_NRF. I think I will open a separate PR for this.

Yes, this is the correct approach in my opinion. Let's go for this.

Reply:
@agross-linaro the CLOCK_CONTROL_NRF5 has been renamed to CLOCK_CONTROL_NRF, I suppose this issue could be resolved?